### ISSUE TITLE
Rust/intellij "unresolved reference"

### ISSUE THREAD

**Issue by zeroexcuses** [intellij-rust/intellij-rust#3116]
1. Consider the following code fragment. It compiles fine:
```rust
pub trait FooT {
    fn foo(&self);
}

pub trait EnvT {
    type SomeType: FooT;
}

pub struct Apple<Env: EnvT> {
    pub some_val: Env::SomeType,
}

impl<Env: EnvT> Apple<Env> {
    pub fn blah(&self) {
        self.some_val.foo();
    }
}
```

2. Now, on IntelliJ + Rust-plugin, I am getting "undefined referenced" (and color red) for the "foo" in `self.some_val.foo`

3. My question: I know the plugin is not complete. If possible, I would like to adjust my programming style to make the plugin happy (to avoid false positives, so that I'll take red-text more seriously).

4. Anyone have advice on how to make the plugin happy in this case?

**Comment by Undin** [comment]
@zeroexcuses Maybe `Ignore unresolved references without quick fix` option of `Unresolved reference` inspection is disabled?
<img width="595" alt="screen shot 2018-12-06 at 21 41 05" src="https://user-images.githubusercontent.com/2539310/49604796-eb717e00-f99f-11e8-9809-ac8a904dea6e.png">
You can enable this option to avoid some false positive `Unresolved reference` error annotation

**Comment by zeroexcuses** [comment]
I am sorry for not being clear. I have "ignore unresolved references" turned OFF. My goal was to program in such a way that all references _could_ be resolved.

Is there a hack workaround? I am willing to slightly change the way I program to keep the plugin happy.

**Comment by vlad20012** [comment]
> I am willing to slightly change the way I program to keep the plugin happy.

I don't think it's reasonable or even possible. We have a lot of known holes in name resolution, type inference and macros support, that results to the fact that a lot of references are not resolved. We are working in it and we will enable this inspection by default when we are sure that it is wok correctly.

For now, if you want to see all unresolved reference errors n your project, you can enable "cargo check" feature in out plugin settings:
![Uploading image.png…]()
